tests/cookie_test.cc

```diff
@@ -129,3 +129,25 @@ TEST(cookie_test, invalid_test) {
     ASSERT_THROW(Cookie::fromString("lang=en-US; Max-Age=12ab"), std::invalid_argument);
 }
 
+void addCookies(const char* str, std::function<void (const CookieJar&)> testFunc) {
+    CookieJar jar;
+    jar.addFromRaw(str, strlen(str));
+    testFunc(jar);
+}
+
+TEST(cookie_test, cookiejar_test) {
+    addCookies("key1=value1", [](const CookieJar& jar) {
+        ASSERT_EQ(jar.get("key1").value, "value1");
+    });
+
+    addCookies("key2=value2; key3=value3; key4=; key5=foo=bar", [](const CookieJar& jar) {
+        ASSERT_EQ(jar.get("key2").value, "value2");
+        ASSERT_EQ(jar.get("key3").value, "value3");
+        ASSERT_EQ(jar.get("key4").value, "");
+        ASSERT_EQ(jar.get("key5").value, "foo=bar");
```

Comment:
This is valid and what I expected, per https://tools.ietf.org/html/rfc6265#section-4.1